addons/account/models/account.py

```diff
@@ -253,6 +256,24 @@ def _check_currency(self):
             if self.default_debit_account_id and not self.default_debit_account_id.currency_id.id == self.currency_id.id:
                 raise UserError(_('Configuration error!\nThe currency of the journal should be the same than the default debit account.'))
 
+    @api.one
```

Comment:
That would help. I don't know enough about the business logic but if possible I would do some refactoring on res.partner.bank :
- drop address fields and keep the res.partner 'Account Holder'
- drop bank_bic and bank_name to just keep bank_id
- rename state in type, make it required with default value 'Domestic Account'

That way, there's just 3 fields to fill in : account type, account number and bank. The partner_id comes from the context (add a bank account for a partner and it's the partner, add a bank account for the journal and it's company.partner_id)